lib/cea/cea_decoder.js

```diff
@@ -211,53 +209,53 @@ shaka.cea.CeaDecoder = class {
     const selectedMode = `CC${(fieldNum << 1) | selectedChannel + 1}`;
     const selectedStream = this.cea608ModeToStream_.get(selectedMode);
 
-    // If on field 1, check if there are too many bad frames
-    if (fieldNum === 0) {
-      if ((b1 === 0xff && b2 === 0xff) || (!b1 && !b2)) {
-        // Completely invalid frame.  FCC says after 45 of them, reset.
-        if (++this.badFrames_ >= 45) {
-          this.reset();
-        }
+    // Check for bad frames (bad pairs). This can be two 0xff, two 0x00, or any
+    // byte of even parity. ccData1 and ccData2 should be uint8 of odd parity.
+    if ((ccPacket.ccData1 === 0xff && ccPacket.ccData2 === 0xff) ||
+        (!ccPacket.ccData1 && !ccPacket.ccData2) ||
+        !this.isOddParity_(ccPacket.ccData1) ||
+        !this.isOddParity_(ccPacket.ccData2)) {
+      // Per CEA-608-B C.21, reset the memory after 45 consecutive bad frames.
+      if (++this.badFrames_ >= 45) {
+        this.reset();
       }
-    }
-
-    // Validate the parity, cc_data_1 and cc_data_2 should be odd parity uimsbf.
-    if (!this.isOddParity_(b1) || !this.isOddParity_(b2)) {
       return null;
     }
+    this.badFrames_ = 0;
 
     // Remove the MSB (parity bit).
-    ccPacket.ccData1 = (b1 &= 0x7f);
-    ccPacket.ccData2 = (b2 &= 0x7f);
+    ccPacket.ccData1 &= 0x7f;
+    ccPacket.ccData2 &= 0x7f;
```

Comment:
Are these packets completely discarded after decode() is complete?  If so, this should be fine.  But if they were ever re-processed, this would cause half your packets to fail the parity check on a second pass.

Reply:
Yes, once the decode is complete the packets are all cleared. There is no reprocessing of packets after.